src/index.ts

```diff
@@ -329,7 +329,10 @@ export class GraphQLServer {
     return new Promise((resolve, reject) => {
       const combinedServer = server
       combinedServer.listen(this.options.port, () => {
-        callbackFunc(this.options)
+        callbackFunc({
+          ...this.options,
+          port: combinedServer.address().port,
```

Comment:
Could you quickly lay out the scenario where the port of `combinedServer` would be different than in `this. options`?

Reply:
Yes, when we provide port `0`, node.js let system determine the random free port, like `12543`:
https://stackoverflow.com/a/28050404/5472617